k-distribution/tutorial/1_k/3_lambda++/lesson_6/lambda.md

```diff
@@ -43,6 +43,7 @@ We move all the LAMBDA++ syntax here.
                | Exp Exp              [strict, left]
                | "(" Exp ")"          [bracket]
 // Arithmetic
+               > "-" Int
```

Comment:
Why did you decide to add unary minus? The original definition didn't have it.
Dito for the other lessons.

Reply:
The reason is that we are importing DOMAINS-SYNTAX instead of DOMAINS, which imports UNSIGNED-INT-SYNTAX instead of INT. As a result, there is no other way of expressing negative integers aside from this production. While it may have seemed to work previously, you would have still ended up with parser errors resulting from the lexer/parser ambiguity the old grammar had. The tests simply didn't exercise this case. So I elected to fix all the definitions so that they appropriately describe negative integers in an unambiguous fashion.